companion/src/simulation/telemetrysimu.cpp

```diff
@@ -1089,3 +1091,226 @@ void TelemetrySimulator::LogPlaybackController::setUiDataValues()
     }
   }
 }
+
+void TelemetrySimulator::on_saveTelemetryvalues_clicked()
+{
+    QString idFileNameAndPath = QFileDialog::getSaveFileName(NULL, tr(".txt File"), g.eepromDir(), tr(".txt Files (*.txt)"));
+    if (idFileNameAndPath.isEmpty())
+        return;
+
+    QFile file(idFileNameAndPath);
+    if (!file.open(QIODevice::WriteOnly)){
+        QMessageBox::information(0,"info",file.errorString());
+        return;
+    }
+    QTextStream out(&file);
+
+    out<< ui -> rxbt -> text();
+    out<<"\r\n";
+    out<< ui -> Rssi -> text();
+    out<<"\r\n";
+    out<< ui -> Swr -> text();
+    out<<"\r\n";
+    out << ui -> A1 -> text();
+    out<<"\r\n";
+    out<< ui -> A2 -> text();
+    out<<"\r\n";
+    out<< ui -> A3 -> text();
+    out<<"\r\n";
+    out << ui -> A4 -> text();
+    out<<"\r\n";
+    out << ui -> T1 -> text();
+    out<<"\r\n";
+    out << ui -> T2 -> text();
+    out<<"\r\n";
+    out << ui -> rpm -> text();
+    out<<"\r\n";
+    out << ui -> fuel -> text();
+    out<<"\r\n";
+    out << ui -> fuel_qty -> text();
+    out<<"\r\n";
+    out << ui -> vspeed -> text();
+    out<<"\r\n";
+    out << ui -> valt -> text();
+    out<<"\r\n";
+    out << ui -> vfas -> text();
+    out<<"\r\n";
+    out << ui -> curr -> text();
+    out<<"\r\n";
+    out << ui -> cell1 -> text();
+    out<<"\r\n";
+    out << ui -> cell2 -> text();
+    out<<"\r\n";
+    out << ui -> cell3 -> text();
+    out<<"\r\n";
+    out << ui -> cell4 -> text();
+    out<<"\r\n";
+    out << ui -> cell5 -> text();
+    out<<"\r\n";
+    out << ui -> cell6 -> text();
+    out<<"\r\n";
+    out << ui -> aspeed -> text();
+    out<<"\r\n";
+    out << ui -> gps_alt -> text();
+    out<<"\r\n";
+    out << ui -> gps_speed -> text();
+    out<<"\r\n";
+    out << ui -> gps_course -> text();
+    out<<"\r\n";
+    out << ui -> gps_time -> text();
+    out<<"\r\n";
+    out << ui -> gps_latlon -> text();
+    out<<"\r\n";
+    out << ui -> accx -> text();
+    out<<"\r\n";
+    out << ui -> accy -> text();
+    out<<"\r\n";
+    out << ui -> accz -> text();
+    out<<"\r\n";
+
+    file.close();
+
+}
+
+
+void TelemetrySimulator::on_loadTelemetryvalues_clicked()
+{
+    QString idFileNameAndPath = QFileDialog::getOpenFileName(NULL, tr(".txt File"), g.eepromDir(), tr(".txt Files (*.txt)"));
+    if (idFileNameAndPath.isEmpty())
+        return;
+
+    QFile file(idFileNameAndPath);
+
+    if (!file.open(QIODevice::ReadOnly)){
+        QMessageBox::information(0,"info",file.errorString());
+        return;
+    }
```

Comment:
Comment as above.

    QString fldr = g.backupDir().trimmed();
    if (fldr.isEmpty())
      fldr = QStandardPaths::writableLocation(QStandardPaths::DocumentsLocation);

    QString idFileNameAndPath = QFileDialog::getOpenFileName(this, tr("Open Telemetry File"), fldr % "/telemetry.txt", tr(".txt Files (*.txt)"));
    if (idFileNameAndPath.isEmpty())
        return;

    QFile file(idFileNameAndPath);

    if (!file.open(QIODevice::ReadOnly)){
        QMessageBox::critical(this, CPN_STR_APP_NAME, tr("Unable to open file for reading.\n%1").arg(file.errorString()));
        return;
    }